packit_service/worker/handlers/open_scan_hub.py

```diff
@@ -93,59 +147,139 @@ def check_scan_and_build(self):
 
-                defects = data.get("defects")
-                if defects is None:
-                    logger.debug("No 'defects' array found in the JSON data.")
-                    return None
+        self.get_helper().report(
+            state=state,
+            description=description,
+            url=get_openscanhub_info_url(self.event.scan.id),
+            links_to_external_services=external_links,
+        )
 
-                return len(defects)
+        return TaskResults(
+            success=True,
+            details={},
+        )
 
-        except requests.exceptions.RequestException as e:
-            logger.error(f"Error while downloading the JSON file: {e}")
-            return None
-        except json.JSONDecodeError:
-            logger.error("The response is not a valid JSON format.")
-            return None
 
-    def get_issues_added_url(
-        self,
-        openscanhub_url: str = "https://openscanhub.fedoraproject.org",
-        file_format: str = "html",
-    ) -> str:
-        """
-        Constructs the URL for the added issues in the specified
-        format for the given OpenScanHub task.
+@configured_as(job_type=JobType.copr_build)
+@reacts_to(openscanhub.task.Started)
+class CoprOpenScanHubTaskStartedHandler(
+    CoprOpenScanHubAbstractHandler,
+):
+    task_name = TaskName.openscanhub_task_started
 
-        Parameters:
-            openscanhub_url (str)
-            file_format (str): The format of the added issues file ('html' or 'json').
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.event: openscanhub.task.Started = self.data.to_event()
 
-        Returns:
-            str: The full URL to access the added issues in the specified format.
-        """
-        return f"{openscanhub_url}/task/{self.event.task_id}/log/added.{file_format}"
+    def run(self) -> TaskResults:
+        self.check_scan_and_build()
+
+        state = BaseCommitStatus.running
+        description = "Scan in OpenScanHub has started."
+        self.event.scan.set_status(OSHScanStatus.running)
+
+        self.get_helper().report(
+            state=state,
+            description=description,
+            url=self.event.scan.url,
+        )
+
+        return TaskResults(
+            success=True,
+            details={},
+        )
+
+
+class KojiOpenScanHubAbstractHandler(OpenScanHubAbstractHandler):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def get_helper(self) -> KojiOpenScanHubHelper:
+        build_helper = KojiBuildJobHelper(
+            service_config=self.service_config,
+            package_config=self.package_config,
+            project=self.project,
+            metadata=self.data,
+            db_project_event=self.data.db_project_event,
+            job_config=self.job_config,
+        )
+
+        return KojiOpenScanHubHelper(
+            koji_build_helper=build_helper,
+            build=self.event.build,
```

Comment:
The constructor expects `KojiBuildTargetModel`, but you are passing this:

https://github.com/packit/packit-service/blob/c0ec4608a763d7fcc7a6b9b50a7457f1098ddddb/packit_service/events/openscanhub/abstract.py#L36

Reply:
It was fixed by using a union.